src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts

```diff
@@ -451,6 +469,38 @@ export class QuizExerciseDetailComponent implements OnInit, OnChanges, Component
         );
     }
 
+    onExamSelect(): void {
+        this.allExistingQuestions = this.existingQuestions = [];
+        if (!this.selectedExamId) {
+            return;
+        }
+
+        /** Search the selected exam by id in all available exams **/
+        const selectedExam = this.exams.find((exam) => exam.id === Number(this.selectedExamId))!;
+
+        // For the given exam, get list of all quiz exercises. And for all quiz exercises, get list of all questions in a quiz exercise
+        this.quizExerciseService.findForExam(selectedExam.id!).subscribe(
```

Comment:
Puh this is a monster. I'd suggest either breaking it into smaller functions or letting the server do the heavy lifting

Reply:
Extracted some logic from the already existing onCourseSelect and on ExamSelect to shorten the methods a bit.